src/api/uploads/MultiputUpload.js

```diff
@@ -0,0 +1,367 @@
+/**
+ * @flow
+ * @file Multiput upload
+ * @author Box
+ */
+
+import Base from '../Base';
+import { getFileLastModifiedAsISONoMSIfPossible } from '../../util/uploads';
+import createWorker from '../../util/uploadsSHA1Worker';
+import MultiputPart, { PART_STATE_UPLOADED, PART_STATE_DIGEST_READY } from './MultiputPart';
+import type { StringAnyMap } from '../../flowTypes';
+
+const DEFAULT_MULTIPUT_CONFIG = {
+    console: false,
+    digestReadahead: 5,
+    initialRetryDelayMs: 5000,
+    maxRetryDelayMs: 60000,
+    parallelism: 5,
+    partSize: null,
+    requestTimeout: 120000,
+    retries: 5
+};
+
+class MultiputUpload extends Base {
+    clientId: ?string;
+    commitRetryCount: number;
+    config: Object;
+    createSessionNumRetriesPerformed: number;
+    destinationFile: ?string;
+    destinationFolder: ?string;
+    ended: boolean;
```

Comment:
What's the difference between this and isDestroyed()? Do we need both?

Reply:
Good catch, removing `ended `